configuration/configuration.go

```diff
@@ -260,6 +261,38 @@ func NewConfigurationData(mainConfigFile string, serviceAccountConfigFile string
 	return c, nil
 }
 
+func (c *ConfigurationData) checkServiceAccountConfig() {
+	notFoundServiceAccountNames := map[string]bool{
+		"fabric8-wit":           true,
+		"fabric8-tenant":        true,
+		"fabric8-jenkins-idler": true,
+		"fabric8-oso-proxy":     true,
+		"online-registration":   true,
+		"fabric8-notification":  true,
+		"rh-che":                true,
+	}
+	for _, sa := range c.sa {
+		if sa.Name == "" {
+			msg := "service account name is empty in service account config"
+			c.appendDefaultConfigErrorMessage(&msg)
```

Comment:
Can't we maintain an array ( named `missingServiceAccounts`  initialized as empty ) where we keep pushing the names which are missing and in the end return an error, if `len(missingServiceAccounts) != 0` .. instead of maintaining `notFoundServiceAccountNames` ?

Reply:
We don't need an array or map to check the number of SAs :) The point is to check if all expected SAs are present in configuration.

Reply:
Understood, this is the only place ( others are example files ) where you can maintain actual service account names, and not call it an example :)